thunder/core/jit_ext.py

```diff
@@ -1608,6 +1623,8 @@ def thunder_general_jit(
             process_recorded_modifications(ctx, epilogue_trace)
             last_interpreter_log = jfn._last_interpreter_log
 
+    computation_trace = functionalize_inplace_ops(computation_trace)
+
```

Comment:
I think it's useful to apply functionalization where possible, let's make this step more discoverable and also add an option to disable it by putting it into the `thunder.jit` function.